### ISSUE TITLE
Flow UI: Component page could show recent logs

### ISSUE THREAD

**Issue by tpaschalis** [grafana/agent#2872]
In the future, the Flow UI could be used to display the recent logs (~5 minutes?) emitted by a component in its details page.

Since Flow has a better story around a shared logging interface, and by default all components log in a different stream, it might be easier to do than expected. We might have to alleviate some concerns around security, but by default the UI is only exposed on localhost so I don't think it's more of a risk than the normal logging in a Kubernetes environment for example.

When debugging an Unhealthy component, the extra hints alongside its configuration and dependents could be useful.

**Comment by mattdurham** [comment]
Assuming we can have some controls around it, this sounds extremely solid idea.

**Comment by rfratto** [comment]
Are you imagining a stream of logs or it only shows the last 5 minutes of logs from the time the page is loaded? 

Also: ~minutes of logs might be an issue for extremely noisy components. What about the last N logs (like 100)? (Or even last X minutes up to N maximum logs?)

**Comment by tpaschalis** [comment]
Good point about noisy components, we'll figure out a way to trim this down.

A stream of logs would that is enabled on demand would be the best, but even a static view of the last N logs kept in a queue would be a good start.
